app/src/main/java/io/github/droidkaigi/confsched2017/viewmodel/SessionViewModel.java

```diff
@@ -117,6 +117,8 @@ private int decideColSpan(@NonNull Session session, int roomCount) {
             return 3;
         } else if (session.isBreak()) {
             return roomCount;
+        } else if (session.isDinner()) {
+            return 6;
```

Comment:
never mind. let me sleep tonight. I will think about this one.

By the way, I totally agree with what you wrote. colspan should be handled in ViewModel.